drivers/digital-io/max22190/iio_max22190.c

```diff
@@ -0,0 +1,544 @@
+*/
+static int max22190_iio_read_fault1(void *dev, char *buf, uint32_t len,
+				    const struct iio_ch_info *channel,
+				    intptr_t priv)
+{
+	struct max22190_iio_desc *desc = dev;
+	uint32_t val;
+	int ret;
+
+	ret = max22190_reg_read(desc->max22190_desc, MAX22190_FAULT1_REG, &val);
+	if (ret)
+		return ret;
+
+	return iio_format_value(buf, len, IIO_VAL_INT, 1, (int32_t *)&val);
+}
+
+/**
+ * @brief Read the fault2 attribute for a specific channel
+ * @param dev - The iio device structure
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor.
+ * @return 0 in case of succes, error code otherwise.
+*/
+static int max22190_iio_read_fault2(void *dev, char *buf, uint32_t len,
+				    const struct iio_ch_info *channel,
+				    intptr_t priv)
+{
+	struct max22190_iio_desc *desc = dev;
+	uint32_t val;
+	int ret;
+
+	ret = max22190_reg_read(desc->max22190_desc, MAX22190_FAULT2_REG, &val);
+	if (ret)
+		return ret;
+
+	return iio_format_value(buf, len, IIO_VAL_INT, 1, (int32_t *)&val);
+}
+
+/**
+ * @brief Register read wrapper
+ * @param dev - The iio device structure.
+ * @param reg - The register's address.
+ * @param readval - Register value.
+ * @return 0 in case of succes, error code otherwise
+*/
+static int max22190_iio_reg_read(struct max22190_iio_desc *dev, uint32_t reg,
+				 uint32_t *readval)
+{
+	return max22190_reg_read(dev->max22190_desc, reg, readval);
+}
+
+/**
+ * @brief Register write wrapper
+ * @param dev - The iio device structure.
+ * @param reg - The register's address.
+ *
+ * @param writeval - Register value.
+ *
+ * @return 0 in case of succes, error code otherwise
+*/
+static int max22190_iio_reg_write(struct max22190_iio_desc *dev, uint32_t reg,
+				  uint32_t writeval)
+{
+	return max22190_reg_write(dev->max22190_desc, reg, writeval);
+}
+
+/**
+ * @brief Configure a set of IIO channels based on the desired channels.
+ * @param desc - MAX22190 IIO device descriptor.
+ * @param ch_enabled - bool array with the requsted channels.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+static int max22190_iio_setup_channels(struct max22190_iio_desc *desc,
+				       bool *ch_enabled)
+{
+	struct iio_channel *max22190_iio_channels;
+	uint32_t enabled_ch = 0;
+	uint32_t ch_offset = 0;
+	uint32_t i;
+
+	for (i = 0; i < MAX22190_CHANNELS; i++) {
+		if (ch_enabled[i])
+			enabled_ch++;
+	}
+
+	max22190_iio_channels = no_os_calloc(enabled_ch,
+					     sizeof(*max22190_iio_channels));
+	if (!max22190_iio_channels)
+		return -ENOMEM;
+
+	for (i = 0; i < MAX22190_CHANNELS; i++) {
+		if (ch_enabled[i]) {
+			max22190_iio_channels[ch_offset] = (struct iio_channel)MAX22190_CHANNEL(i);
+			max22190_iio_channels[ch_offset].attributes = max22190_attrs;
+			max22190_iio_channels[ch_offset].ch_out = 0;
+			ch_offset++;
+		}
+	}
```

Comment:
Is there any reason why wouldn't all channels be enabled?

Reply:
No, will remove the boolean array and everything related to it.